src/functional/list.h

```diff
@@ -0,0 +1,31 @@
+#ifndef LIST_H
+#define LIST_H
+
+#include <stddef.h>
+#include <stdbool.h>
+
+typedef struct list_t list_t;
+
+struct list_t {
+	void* head;
+	void* tail;
```

Comment:
sure thing.
Due to recursive definition there is no distinction between list and its node i.e. created node is a list by itself.

take for example Haskell definition
```hs
data LinkedList a = Empty | Cons a (LinkedList a) 
{- here LinkedList is our type and Empty and Cons are value constructors -}
```
analogy in C would be
```c
struct linked_list {
  void* head;
  struct linked_list* tail;
};

/* where Empty and Cons constructors replaced by NULL and cons respectively */
```